openwisp_controller/subnet_division/rule_types/base.py

```diff
@@ -65,6 +65,13 @@ def should_create_subnets_ips(cls, instance, **kwargs):
         """
         raise NotImplementedError()
 
+    @classmethod
+    def provision_for_existing_objects(cls, rule_obj):
+        """
+        Contains logic to trigger provisioning for existing objects
+        """
+        pass
```

Comment:
why not raise `NotImplementedError` here?

Reply:
The new rule need not apply on existing objects (devices). Raising `NotImplementedError` will make users implement this method even when no operation is required to be performed. Hence, I used `pass` for this one.

Reply:
I think in most cases we would need that method to be implemented (also for consistency). Cases in which it should not be implemented would be an exception rather than the rule so hence why I think raising NotImplementedError is useful.